link.py

```diff
@@ -0,0 +1,115 @@
+from collections import deque
+'''
+   Class for a link.
+   
+   Attributes:
```

Comment:
New indentation level after attributes. Also format for attributes should be like:
link_id (int): a unique identifier for each link